server/graphql/v2/object/Host.js

```diff
@@ -1232,12 +1232,34 @@ export const GraphQLHost = new GraphQLObjectType({
           }
 
           if (args.isUnhosted) {
-            const unhostedActivities = await models.Activity.findAll({
-              where: { type: 'collective.unhosted', HostCollectiveId: host.id },
+            const formerHostedMembers = await models.Member.findAll({
+              where: { MemberCollectiveId: host.id, role: 'HOST', deletedAt: { [Op.ne]: null } },
+              include: [
+                {
+                  model: models.Collective,
+                  as: 'collective',
+                  required: true,
+                  attributes: ['id'],
+                  include: [
+                    {
+                      model: models.HostApplication,
+                      as: 'hostApplications',
+                      required: true,
+                      attributes: ['id'],
+                      where: {
+                        status: 'APPROVED',
+                        HostCollectiveId: host.id,
+                      },
```

Comment:
Note that this won't cover everything: when we introduced host applications in https://github.com/opencollective/opencollective-api/pull/5002 (3 years ago), we didn't create host applications for already hosted collectives. I do believe it's a good idea to look at this table, though; it's the right place to trust for this kind of info.

If we want it to be reliable, I would suggest a migration to insert entries for existing collectives. Something like:

```sql
INSERT INTO "HostApplications" ("CollectiveId", "HostCollectiveId", "status", "createdAt")
VALUES (collective.id, collective."HostCollectiveId", 'APPROVED', COALESCE(collective."approvedAt", collective."createdAt"))
FROM "Collectives" collective
WHERE collective."isActive" IS TRUE
AND collective."deletedAt" IS NULL
AND collective."HostCollectiveId" IS NOT NULL
AND collective.id != collective."HostCollectiveId"
```

Reply:
Hey @Betree, there's something intrinsically off with this logic since not every collective needs to go through the HostApplications path in order to get hosted, i.e. automated GitHub stars approval, collective created by a fiscal host admin, etc...

Perhaps going through the other way around? Deleted host members that do not have a denied HostAaplication?

Reply:
True.

Looking at deleted members that do not have a denied HostAplication could work but make sure to only consider the last one (a collective could be rejected, then accepted).

An alternative would be to look at the `CollectiveHistories`; the query would be super simple and reliable, but incredibly slow. We could add an index to resolve that, but altogether I don't think that would be a good solution.

Reply:
> Looking at deleted members that do not have a denied HostAplication could work but make sure to only consider the last one (a collective could be rejected, then accepted).

We're also making sure HostCollectiveId does not match, which covers this edge case.

Reply:
> > Looking at deleted members that do not have a denied HostAplication could work but make sure to only consider the last one (a collective could be rejected, then accepted).
> 
> We're also making sure HostCollectiveId does not match, which covers this edge case.

Not exactly. The case I'm referring to is:
1. Collective applies to host
2. Host rejects the application: "please include blablabla..."
3. Collective re-applies with the requested info
4. Application gets approved

There are now two applications in the DB for this collective: a rejected one, and an approved one.

Reply:
That's not a problem if the `Collective.HostCollectiveId` != host id. The purpose of the tab is displaying all collectives that were once hosted, not the ones that were unhosted only after having the latest application approved.